cloud/pod_agent_util.go

```diff
@@ -7,29 +7,42 @@ import (
 
 	"github.com/evergreen-ci/evergreen"
 	"github.com/evergreen-ci/evergreen/model/pod"
-	"github.com/pkg/errors"
 )
 
 // agentScript returns the script to provision and run the agent in the pod's
 // container.
-func agentScript(settings *evergreen.Settings, p *pod.Pod) ([]string, error) {
-	downloadCmd, err := downloadAgentCommands(settings, p)
-	if err != nil {
-		return nil, errors.WithStack(err)
+func agentScript(settings *evergreen.Settings, p *pod.Pod) []string {
+	scriptCmds := []string{downloadAgentCommands(settings, p)}
+	if p.TaskContainerCreationOpts.OS == pod.OSLinux {
+		scriptCmds = append(scriptCmds, fmt.Sprintf("chmod +x %s", clientName(p)))
 	}
-
 	agentCmd := strings.Join(agentCommand(settings, p), " ")
+	scriptCmds = append(scriptCmds, agentCmd)
 
-	return []string{
-		"bash", "-c",
-		strings.Join([]string{downloadCmd, fmt.Sprintf("chmod +x %s", clientName(p)), agentCmd}, " && "),
-	}, nil
+	return append(invokeShellScriptCommand(p), strings.Join(scriptCmds, " && "))
 }
 
-// agentCommand returns the arguments to start the agent on a pod.
+// invokeShellScriptCommand returns the arguments to invoke an in-line shell
+// script in the pod's container.
+func invokeShellScriptCommand(p *pod.Pod) []string {
+	if p.TaskContainerCreationOpts.OS == pod.OSWindows {
+		return []string{"cmd.exe", "/c"}
+	}
+
+	return []string{"bash", "-c"}
+}
+
+// agentCommand returns the arguments to start the agent in the pod's container.
 func agentCommand(settings *evergreen.Settings, p *pod.Pod) []string {
+	var pathSep string
+	if p.TaskContainerCreationOpts.OS == pod.OSWindows {
+		pathSep = "\\"
+	} else {
+		pathSep = "/"
+	}
+
 	return []string{
-		fmt.Sprintf("./%s", clientName(p)),
+		fmt.Sprintf(".%s%s", pathSep, clientName(p)),
 		"agent",
 		fmt.Sprintf("--api_server=%s", settings.ApiUrl),
 		fmt.Sprintf("--mode=pod"),
```

Comment:
Should the file paths be joined with the window's path separator when on windows?

Reply:
That's already what the code on [L39](https://github.com/evergreen-ci/evergreen/pull/5015/files#diff-7be997f174f3941ca470d8e8500ac29af7bdbb5452b19d47d2905c00a4830dadR39) does.